plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,614 @@
+                            description: The maximum number of parts allowed in the response.
+                            returned: always
+                            type: int
+                        is_truncated:
+                            description: Indicates whether the returned list of parts is truncated.
+                            returned: always
+                            type: bool
+                storage_class:
+                    description: The storage class information of the object.
+                    returned: always
+                    type: str
+                    sample: "STANDARD"
+                parts:
+                    description: A container for elements related to an individual part.
+                    returned: always
+                    type: complex
+                    contains:
+                        part_number:
+                            description: The part number identifying the part. This value is a positive integer between 1 and 10,000.
+                            returned: always
+                            type: int
+                        size:
+                            description: The size of the uploaded part in bytes.
+                            returned: always
+                            type: int
+                        checksum_crc32:
+                            description: The base64-encoded, 32-bit CRC32 checksum of the object.
+                            returned: if it was upload with the object.
+                            type: str
+                            sample: "xxxxxxxxxxxx"
+                        checksum_crc32c:
+                            description: The base64-encoded, 32-bit CRC32C checksum of the object.
+                            returned: if it was upload with the object.
+                            type: str
+                            sample: "xxxxxxxxxxxx"
+                        checksum_sha1:
+                            description: The base64-encoded, 160-bit SHA-1 digest of the object.
+                            returned: if it was upload with the object.
+                            type: str
+                            sample: "xxxxxxxxxxxx"
+                        checksum_sha256:
+                            description: The base64-encoded, 256-bit SHA-256 digest of the object.
+                            returned: if it was upload with the object.
+                            type: str
+                            sample: "xxxxxxxxxxxx"
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
```

Comment:
It would be helpful if this module **did** support listing all the objects in a bucket, that would mean we could deprecate ``state=list`` over in s3_object

Reply:
Added support to replicate behavior of `s3_object` `state=list`.